specs/hotel_spec.rb

```diff
@@ -0,0 +1,242 @@
+require_relative 'spec_helper'
+# require 'pry'
+describe 'Hotel' do
+  before do
+    @hotel = Hotel::Hotel.new
+    @date1 = Date.new(2018,9,12)
+    @date2 = Date.new(2018,9,15)
+    @date3 = Date.new(2018,9,13)
+    @date4 = Date.new(2018,9,19)
+  end
+
+  describe 'initialize' do
+    it 'creates an instance of Hotel' do
+      @hotel.must_be_instance_of Hotel::Hotel
+    end
+
+    it 'creates a hash of rooms' do
+      @hotel.rooms.must_be_kind_of Array
+    end
+
+    it 'creates an array of reservations' do
+      @hotel.reservations.must_be_kind_of Array
+    end
+
+  end
+
+  describe 'make_reservation' do
+    it 'creates a reservation object with two dates and pushes to reservations' do
+      @hotel.make_reservation(date1: @date1, date2: @date2).must_be_kind_of Array
+    end
+    it 'creates a reservation object with one date and pushes to reservations' do
+      @hotel.make_reservation(date1: @date1).must_be_kind_of Array
+    end
+    it 'returns an array of reservation objects' do
+      @hotel.reservations.each do |reservation|
+        reservation.must_be_instance_of Hotel::Reservation
+      end
+    end
```

Comment:
You should test your error handling here. What happens if there aren't any rooms available?